README.md

```diff
@@ -29,122 +37,1058 @@ The library design is inspired by the work in [Exploring HTN Planners through ex
+
+We hope this also explains why the name `View` was chosen to pass [references to effect and action functions](#basic-usage), changes to the view are indeed propagated to the global state object.
+
+We can reuse these tasks in methods the same way as before, but we need to pass the values for the placeholder variables along with the target.
+
+```typescript
+const plusTwo = Task.of<System>().from({
+	lens: '/counters/:counterId',
+	condition: (counter, { target }) => target - counter > 1,
+	method: (_, { target, counterId }) => [
+		plusOne({ target, counterId }),
+		plusOne({ target, counterId }),
+	],
+	description: ({ counterId }) => `${counterId} + 2`,
+});
+```
+
+## Parallelism
+
+What happens with the above case if we look for a plan for multiple counters? What will be the plan generated below?
+
+```typescript
+const planner = Planner.from({
+	tasks: [plusOne],
+});
+
+// Find a plan from 0 to 3
+const res = planner.findPlan(
+	{ counters: { a: 0, b: 0 } },
+	{ counters: { a: 2, b: 2 } },
+);
+```
+
+Again we can draw the plan using the mermaid test helper
+
+```mermaid
+graph TD
+        start(( ))
+        start -.- d0{ }
+        d0 -.- 4418a5c("a + 1")
+        4418a5c -.- d1{ }
+        d1 -.- a53644f("a + 1")
+        a53644f -.- d2{ }
+        d2 -.- 47bcbb6("b + 1")
+        47bcbb6 -.- d3{ }
+        d3 -.- a48a56b("b + 1")
+        a48a56b -.- stop(( ))
+        stop:::finish
+        classDef finish stroke:#000,fill:#000
+        start:::selected
+        start --> 4418a5c
+        4418a5c:::selected
+        4418a5c --> a53644f
+        a53644f:::selected
+        a53644f --> 47bcbb6
+        47bcbb6:::selected
+        47bcbb6 --> a48a56b
+        a48a56b:::selected
+        a48a56b --> stop
+        classDef error stroke:#f00
+        classDef selected stroke:#0f0
+```
+
+Even though the `a + 1` and `b + 1` actions modify totally different parts of the state, the planner still comes up with a sequential plan. This is a limitation of the planner right now, that we intend to improve in the future. Not all is bad news though, the planner does support parallel planning for methods. Let's define a generic `nPlusOne` task that increases any pending counters.
+
+```typescript
+const nPlusOne = Task.of<System>().from({
+	lens: '/counters',
+	condition: (counters, { target }) =>
+		Object.keys(counters).some((k) => counters[k] < target[k]),
+	method: (counters, { target }) =>
+		Object.keys(counters).map((k) =>
+			plusOne({ counterId: k, target: target[k] }),
+		),
```

Comment:
So, if am following this right, the resulting method will include a `plusOne` to each of the counters. What if some of them is already at the target? I guess this will simply let the `plusOne` condition do its job and no issues will arise. Or maybe I missed some point?

Regardless, this example was more eyebrow-raising than the others.

Reply:
It is missing a filter, nice catch

Reply:
This still works as the condition in the `plusOne` action does its job, otherwise [this test](https://github.com/balena-io-modules/mahler/blob/5d748808cdb68f15fb718c24cfcaa79176372fce/tests/readme/examples.spec.ts#L407-L415) would fail, but the method is indeed missing a filter